lib/cocoapods/target/aggregate_target.rb

```diff
@@ -7,6 +7,9 @@ class AggregateTarget < Target
     #         generated this target.
     attr_reader :target_definition
 
+    # Product types where the product's frameworks must be embedded in a host target
+    EMBED_FRAMEWORKS_IN_HOST_TARGET_TYPES = [:app_extension, :watch_extension].freeze
```

Comment:
Got it. Thanks. I can do this, but I'll need to update Xcodeproj again to remove the extension target restriction. `host_targets_for_extension_target ` raises if the passed target is not an extension target (will also update the name to be more generic as well). I'll push an update responding to the rest of the feedback I've gotten so far (many thanks for being so responsive!), and then I'll push another update once the Xcodeproj PR is update.